test/tcpReconnect.js

```diff
@@ -85,3 +88,48 @@ test('tcp reconnect', function testTcpReconnect (done) {
     )
   }
 })
+
+test('tcp reconnect after initial failure', async function testTcpReconnectAfterInitialFailure () {
+  let failureCount = 0
+  let counter = 0
+  function sendData () {
+    setInterval(() => {
+      counter++
+      tcpConnection.write(`log${counter}\n`, 'utf8', () => { /* ignore */ })
+    }, 100)
+  }
+  const port = await getPort()
+  const address = '127.0.0.1'
+  const tcpConnection = TcpConnection({
+    address,
+    port,
+    reconnect: true
+  })
+  tcpConnection.on('error', () => { failureCount++ })
+  sendData()
+  const received = await new Promise((resolve, reject) => {
+    let closing = false
+    const received = []
+    setTimeout(() => {
+      const server = startServer({
+        address,
+        port,
+        next: (msg) => {
+          switch (msg.action) {
+            case 'data':
+              received.push(msg)
+              if (!closing) {
+                closing = true
+                server.close(() => {
+                  resolve(received)
+                })
+              }
+          }
+        }
+      })
```

Comment:
Avoid using timers but instead use logical events.

Reply:
I've removed the timer since we don't need it. When we instantiate `TcpConnection` it will try to connect and fail because the server has not yet started (i.e., we don't need to wait for `tcpConnection.write`).